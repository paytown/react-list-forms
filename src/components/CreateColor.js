import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class CreateColor extends PureComponent {
  static propTypes = {
    addColor: PropTypes.func.isRequired
  };

  state = {
    name: '',
    color: '#FF0000'
  }

  handleSubmit = event => {
    event.preventDefault();
    const { name, color } = this.state;
    this.props.addColor({ name, hex: color });
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { color, name } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <input type="text" name="name" value={name} onChange={this.handleChange} />
        <input type="color" name="color" value={color} onChange={this.handleChange} />
        <button>Clicky</button>
      </form>
    );
  }
}
