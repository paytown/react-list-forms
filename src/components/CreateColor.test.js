import React from 'react';
import { shallow } from 'enzyme';
import CreateColor from './CreateColor';

describe('color picker', () => {
  it('renders color', () => {
    const wrapper = shallow(<CreateColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
