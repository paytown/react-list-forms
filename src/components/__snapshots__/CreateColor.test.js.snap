// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color picker renders color 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="#FF0000"
  />
  <button>
    Clicky
  </button>
</form>
`;
